weaviate/graphql/multitargetbuilder.go

```diff
@@ -9,16 +9,17 @@ import (
 )
 
 type MultiTargetArgumentBuilder struct {
-	targetCombination *dto.TargetCombination
+	targetCombination *dto.TargetCombinationType
 	targetVectors     []string
+	weights           [][]float32
 }
 
 func (m *MultiTargetArgumentBuilder) getCombinationMethod() string {
 	combinationMethod := ""
 	if m.targetCombination == nil {
 		return combinationMethod
 	}
-	switch m.targetCombination.Type {
+	switch *m.targetCombination {
 	case dto.Sum:
 		combinationMethod = "sum"
 	case dto.Average:
```

Comment:
Nit: because `TargetCombinationType` is its own type you could move the `switch` to statement into its `String() string` method. 

```go
func (t TargetCombinationType) String() string {
  switch t {
    case Sum:
      return "sum"
   // ...
   }
}
```

Assuming they have constant mapping, i.e. `Sum` should always translate to `"sum"` and so on.

Reply:
that is imported from weaviate and we do not need it there. In general I like the proposal, but I think in this case we should leave it like it is

Reply:
Agreed 👍 

---

Btw, if we still wanted to have `String()` method handle it without cluttering `weaviate` core, we could alias the type:

```go
// In weaviate-go-client
import "github.com/weaviate/weaviate"

type TargetCombinationType weaviate.TargetCombinationType
func (t TargetCombinationType) String() string { ... }
```